staging/src/k8s.io/apimachinery/pkg/util/strategicpatch/patch.go

```diff
@@ -58,11 +62,17 @@ type DiffOptions struct {
 	IgnoreChangesAndAdditions bool
 	// IgnoreDeletions indicates if we keep the deletions in the patch.
 	IgnoreDeletions bool
+	// We introduce a new value replaceKeys for patchStrategy.
+	// It indicates that all fields needing to be preserved must be
+	// present in the `replaceKeys` list.
+	// And the fields that are present will be merged with live object.
+	// All the missing fields will be cleared when patching.
+	ReplaceKeys bool
 }
 
 type MergeOptions struct {
-	// MergeDeleteList indicates if we are merging the delete parallel list.
-	MergeDeleteList bool
+	// MergeParallelList indicates if we are merging the parallel list.
```

Comment:
We should add comments about when this is set / not set in the 3-way and 2-way and why